### PR TITLE
simplifying remove render targets in more C++11 style

### PR DESCRIPTION
**Pull request by Joilnen**

### CODE REVIEW

File: OgreMain/src/OgreRenderSystem.cpp [hunk 1]
```diff
@@ -630,22 +630,15 @@ namespace Ogre {
         // Remove all the render targets. Destroy primary target last since others may depend on it.
         // Keep mRenderTargets valid all the time, so that render targets could receive
         // appropriate notifications, for example FBO based about GL context destruction.
-        RenderTarget* primary = 0;
-        for (RenderTargetMap::iterator it = mRenderTargets.begin(); it != mRenderTargets.end(); /* note - no increment */)
-        {
-            RenderTarget* current = it->second;
-            if (!primary && current->isPrimary())
```
Comment (paroj) [inline OgreMain/src/OgreRenderSystem.cpp:637]: I think you should keep the condition. While there should be only one primary window, it is not strictly enforced.

This way we will at least not leak memory if things go wrong.

**Review by paroj — CHANGES_REQUESTED** [review]

**Review by Joilnen — COMMENTED** [review]
keeping primary like the original code in a second commit

**Review by Joilnen — COMMENTED** [review]
changed

**Review by Joilnen — COMMENTED** [review]
to review

**Pull request merged**